eva-accession-import/src/test/java/uk/ac/ebi/eva/accession/dbsnp/io/DbsnpClusteredVariantWriterTest.java

```diff
@@ -0,0 +1,149 @@
+                                                                 VARIANT_TYPE, VALIDATED);
+        DbsnpClusteredVariantEntity variant = new DbsnpClusteredVariantEntity(EXPECTED_ACCESSION,
+                                                                              hashingFunction.apply(clusteredVariant),
+                                                                              clusteredVariant);
+
+        dbsnpClusteredVariantWriter.write(Collections.singletonList(variant));
+
+        List<DbsnpClusteredVariantEntity> accessions = mongoTemplate.find(new Query(),
+                                                                          DbsnpClusteredVariantEntity.class);
+        assertEquals(1, accessions.size());
+        assertEquals(EXPECTED_ACCESSION, (long) accessions.get(0).getAccession());
+
+        assertEquals(clusteredVariant, accessions.get(0).getModel());
+    }
+
+    @Test
+    public void saveDifferentTaxonomies() throws Exception {
+        ClusteredVariant firstClusteredVariant = new ClusteredVariant("assembly", TAXONOMY_1, "contig", START_1,
+                                                                      VARIANT_TYPE, VALIDATED);
+        ClusteredVariant secondClusteredVariant = new ClusteredVariant("assembly", TAXONOMY_2, "contig", START_1,
+                                                                       VARIANT_TYPE, VALIDATED);
+        DbsnpClusteredVariantEntity firstVariant = new DbsnpClusteredVariantEntity(
+                EXPECTED_ACCESSION, hashingFunction.apply(firstClusteredVariant), firstClusteredVariant);
+        DbsnpClusteredVariantEntity secondVariant = new DbsnpClusteredVariantEntity(
+                EXPECTED_ACCESSION_2, hashingFunction.apply(secondClusteredVariant), secondClusteredVariant);
+
+        dbsnpClusteredVariantWriter.write(Arrays.asList(firstVariant, secondVariant));
+
+        List<DbsnpClusteredVariantEntity> accessions = mongoTemplate.find(new Query(),
+                                                                          DbsnpClusteredVariantEntity.class);
+        assertEquals(2, accessions.size());
+        assertEquals(EXPECTED_ACCESSION, (long) accessions.get(0).getAccession());
+        assertEquals(EXPECTED_ACCESSION_2, (long) accessions.get(1).getAccession());
+
+        assertEquals(firstClusteredVariant, accessions.get(0).getModel());
+        assertEquals(secondClusteredVariant, accessions.get(1).getModel());
+    }
+
+    @Test
+    public void failsOnDuplicateVariant() throws Exception {
+        ClusteredVariant clusteredVariant = new ClusteredVariant("assembly", TAXONOMY_1, "contig", START_1,
+                                                                 VARIANT_TYPE, VALIDATED);
+        DbsnpClusteredVariantEntity variant = new DbsnpClusteredVariantEntity(
+                EXPECTED_ACCESSION, hashingFunction.apply(clusteredVariant), clusteredVariant);
+
+        thrown.expect(RuntimeException.class);
```

Comment:
I would prefer using the `expect` in the annotation as `@Test(expected = RuntimeException.class)`. With that we know easily which test cases are expected to throw an exception

Reply:
I won't open this discussion again :) there's more than one statement in the test, so I'll keep it as-is